test/unittest/StatisticsBackend/StatisticsBackendTests.cpp

```diff
@@ -16,13 +16,208 @@
 
 #include <StatisticsBackend.hpp>
 #include <types/types.hpp>
+#include "database/database.hpp"
 
 #include "gtest/gtest.h"
 
+#include <DatabaseUtils.hpp>
+#include <fastdds-statistics-backend/types/JSONTags.h>
+
 using namespace eprosima::statistics_backend;
+using namespace eprosima::statistics_backend::database;
+
+/**
+ * @brief Fixture for the statistics_backend class tests
+ *
+ * \c get_type Returns the entity kind of a given id.
+ *
+ */
+class statistics_backend_tests : public ::testing::Test
+{
+public:
+
+    typedef uint32_t TestId;
+
+    void SetUp()
+    {
+        entities = PopulateDatabase::populate_database(db);
+    }
+
+    DataBaseTest db;
+    std::map<TestId, std::shared_ptr<const Entity>> entities;
+};
+
+class StatisticsBackendTest : public StatisticsBackend
+{
+public:
+
+    static void set_database(
+            Database* db)
+    {
+        database_ = db;
+    }
+
+    static const char* get_entity_kind_str(
+            EntityKind kind)
+    {
+        return entity_kind_str[(int)kind];
+    }
+
```

Comment:
This would not be necessary if we move the valid values to another header file as a defined constant.